sdk/go/pulumi/types.go

```diff
@@ -419,11 +419,26 @@ func (o *OutputState) ApplyTWithContext(ctx context.Context, applier interface{}
 	return result
 }
 
-// isSecret returns a bool representing the secretness of the Output
-func (o *OutputState) isSecret() bool {
+// IsSecret returns a bool representing the secretness of the Output
+func (o *OutputState) IsSecret() bool {
 	return o.getState().secret
 }
 
+// Unsecret will unwrap a secret output as a new output with a resolved value and no secretness
+func Unsecret(input Output) Output {
+	return UnsecretWithContext(context.Background(), input)
+}
+
+// UnsecretWithContext will unwrap a secret output as a new output with a resolved value and no secretness
+func UnsecretWithContext(ctx context.Context, input Output) Output {
+	var x bool
+	x = false
+	o := toOutputWithContext(ctx, input, &x)
+	// set immediate secretness ahead of resolution/fufillment
```

Comment:
```suggestion
	// set immediate secretness ahead of resolution/fulfillment
```

Reply:
Nice find - I'll merge this and fix that in my next PR